### PR TITLE
fix(expansion-panel): make header interaction cancelable - 9.1.x

### LINKED ISSUE #8190: onInteraction event of the igxExpansionPanelHeaderComponent is not cancelable
## Description  
In regards to a customer requirement of being able to toggle the panel by clicking on the expand/collapse indicator exclusively determined that this event can not be cancelled, which is the best approach for achieving such behavior.

 * igniteui-angular version: 9.1.x, 10.1.x
 * browser: any

## Steps to reproduce  

1. Run this sample: https://stackblitz.com/edit/expansion-panel-cancelable-on-interaction-event
2. Click on the expand indicator
3. Click on the panel`s header

## Result  
Expansion panel is toggled via both expand indicator and panel`s header.

## Expected result  
To have a mechanism for triggering toggling only from the expansion indicator.

### PR DESCRIPTION
**Pull request by ViktorSlavov**
Closes #8190   

### Additional information (check all that apply):
 - [ ] Bug fix
 - [x] New functionality
 - [x] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [x] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [x] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [x] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/expansion-panel/expansion-panel.common.ts [hunk 1]
```diff
@@ -23,3 +23,5 @@ export interface IExpansionPanelEventArgs extends IBaseEventArgs {
     event: Event;
     panel: IgxExpansionPanelBase;
```
Comment (wnvko) [inline projects/igniteui-angular/src/lib/expansion-panel/expansion-panel.common.ts:24]: please change `panel` to `owner`, as it is coming from `IBaseEventArgs`.
Then you can deprecate `panel`.

File: projects/igniteui-angular/src/lib/expansion-panel/expansion-panel-header.component.ts [hunk 2]
```diff
@@ -185,7 +185,11 @@ export class IgxExpansionPanelHeaderComponent {
             evt.stopPropagation();
             return;
          }
-         this.onInteraction.emit({ event: evt, panel: this.panel });
+         const eventArgs = { event: evt, panel: this.panel, cancel: false };
```
Comment (wnvko) [inline projects/igniteui-angular/src/lib/expansion-panel/expansion-panel-header.component.ts:188]: add type here
```suggestion
         const eventArgs: IExpansionPanelInteractionEventArgs = { event: evt, panel: this.panel, cancel: false };
```

File: projects/igniteui-angular/src/lib/expansion-panel/expansion-panel-header.component.ts [hunk 3]
```diff
@@ -194,17 +198,25 @@ export class IgxExpansionPanelHeaderComponent {
     @HostListener('keydown.Alt.ArrowDown', ['$event'])
     public openPanel(event: KeyboardEvent) {
         if (event.altKey) {
+            const eventArgs = { event, panel: this.panel, cancel: false };
```
Comment (wnvko) [inline projects/igniteui-angular/src/lib/expansion-panel/expansion-panel-header.component.ts:201]: add type here:
```suggestion
            const eventArgs: IExpansionPanelInteractionEventArgs = { event, panel: this.panel, cancel: false };
```

File: projects/igniteui-angular/src/lib/expansion-panel/expansion-panel-header.component.ts [hunk 4]
```diff
@@ -194,17 +198,25 @@ export class IgxExpansionPanelHeaderComponent {
...
             this.panel.expand(event);
-            this.onInteraction.emit({ event: event, panel: this.panel });
         }
      }
 
      /** @hidden @internal */
      @HostListener('keydown.Alt.ArrowUp', ['$event'])
      public closePanel(event: KeyboardEvent) {
         if (event.altKey) {
+            const eventArgs = { event, panel: this.panel, cancel: false };
```
Comment (wnvko) [inline projects/igniteui-angular/src/lib/expansion-panel/expansion-panel-header.component.ts:214]:
```suggestion
            const eventArgs: IExpansionPanelInteractionEventArgs = { event, panel: this.panel, cancel: false };
```

File: projects/igniteui-angular/src/lib/expansion-panel/expansion-panel-header.component.ts [hunk 5]
```diff
@@ -131,7 +131,7 @@ export class IgxExpansionPanelHeaderComponent {
      * ```
      */
     @Output()
-    public onInteraction = new EventEmitter<IExpansionPanelEventArgs>();
+    public onInteraction = new EventEmitter<IExpansionPanelInteractionEventArgs>();
```
Comment (wnvko) [inline projects/igniteui-angular/src/lib/expansion-panel/expansion-panel-header.component.ts:134]: you should update the comment too, as it is pointing to the old event arguments type

File: projects/igniteui-angular/src/lib/expansion-panel/expansion-panel.common.ts [hunk 6]
```diff
@@ -23,3 +23,5 @@ export interface IExpansionPanelEventArgs extends IBaseEventArgs {
...
 }
+
+export interface IExpansionPanelInteractionEventArgs extends IExpansionPanelEventArgs, CancelableEventArgs {}
```
Comment (wnvko) [inline projects/igniteui-angular/src/lib/expansion-panel/expansion-panel.common.ts:27]: IMO `ExpansionPanelCancelabelEventArgs` is better naming here.
Additionally we can add `collapsing` and `expanding` event in future, where we can use same type for the event arguments.
Also in most places we are calling cancelable event arguments `Something`**Cancelable**`EventArgs`

**Review by wnvko — CHANGES_REQUESTED** [review]

**Comment by ViktorSlavov** [comment]
@wnvko thanks for the review!
Since this is a feature, I'll drop this PR and continue w/ the one [targeting the newest version](https://github.com/IgniteUI/igniteui-angular/pull/8268)

**Pull request closed without merging**